homeassistant/components/plex/server.py

```diff
@@ -337,7 +337,7 @@ def connect_to_resource(resource):
                 _LOGGER.debug("Photo session detected, skipping: %s", session)
                 continue
 
-            session_username = session.usernames[0]
+            session_username = getattr(session, "usernames", [])[0]
```

Comment:
This will cause an `IndexError` exception.

Reply:
Ugh, you're right. Good catch.

Reply:
Oops, shame on me

Reply:
I'm the one that called it "trivial" 😂

Reply:
Maybe like this?
```suggestion
            session_username = next(iter(getattr(session, "usernames", [])), None)
```

Reply:
I'm digging into reproducing this and seeing if this would be better to fix in the library instead.

Reply:
That might at least be cleaner than my suggestion. ;)

Reply:
Either that or keep the original line but wrap it in a try and catch `(AttributeError, IndexError)`. Both feel gross compared to potentially handling in the library.

Reply:
Use `hasattr`

Reply:
I was finally able to reproduce and this should definitely be fixed in the underlying library. Will open a different PR when that's ready.